Tests/UnitTests/Reporters/DataTableReporterTest.jsl

```diff
@@ -0,0 +1,64 @@
+Names default to here(0);
+
+DataTableReporterTests = ut test case("DataTableReporter") 
+  << Setup(Expr(
+    reporter = ut data table reporter();
+  ));
+  
+ut test(DataTableReporterTests, "FactoryFunction", Expr(
+  ut assert value(reporter, ut instance of("UtDataTableReporter"));
+));
+
+ut test(DataTableReporterTests, "Failure has 1 row", Expr(
+  reporter << Add Failure({"", "", "label", .}, "test expr", "description", "mismatch", Empty());
+  ut assert value(reporter << N successes,0);
+  ut assert value(reporter << N failures,1);
+  ut assert value(reporter << N unexpected throws,0);
+  ut assert value(reporter << N total failures,1);
+));
+
+ut test(DataTableReporterTests, "2 Unexpected Throw has 2 rows", Expr(
+  reporter << Add Unexpected Throw({"", "", "label", .}, "test expr", "description", "mismatch", Empty());
+  reporter << Add Unexpected Throw({"", "", "label", .}, "test expr", "description", "mismatch", Empty());
+  ut assert value(reporter << N successes,0);
+  ut assert value(reporter << N failures,0);
+  ut assert value(reporter << N unexpected throws,2);
+  ut assert value(reporter << N total failures,2);
+));
+
+ut test(DataTableReporterTests, "Success has 1 row", Expr(
+  reporter << Add Success({"", "", "label", .}, "test expr", "description", Empty());
+  ut assert value(reporter << N successes,1);
+  ut assert value(reporter << N failures,0);
+  ut assert value(reporter << N unexpected throws,0);
+  ut assert value(reporter << N total failures,0);
+));
+
+// I don't know how to write this test.
```

Comment:
From here down errors when ran recursively.  Not sure how fix..

Reply:
This works for me once a couple fixes have been done to the reporter (I'll leave comments on those).

```
ut test(DataTableReporterTests, "Returns reference", Expr(
	dtref = reporter << Get Table Reference;
	columncount = length( dtref << Get Column Names("string") );
	ut assert value(columncount , ut greater than( 4 ));
));
  
ut test(DataTableReporterTests, "Table is invisible by default", Expr(
	dtref = reporter << Get Table Reference;
	ut assert value(dtref << has data view(), 0);
));

ut test(DataTableReporterTests2, "Table is visible when requested", Expr(
	dtref = reporter << New Data View
	ut assert value(dtref << has data view(), 1);
));
```

Reply:
Thank you @emccorkle, mistakes are so obvious once someone points them out!